ui/src/components/Settings/Notifications/index.tsx

```diff
@@ -0,0 +1,192 @@
+import {
+  DocumentAddIcon,
+  PlusCircleIcon,
+  TrashIcon,
+} from '@heroicons/react/solid'
+import { useContext, useEffect, useState } from 'react'
+import { useToasts } from 'react-toast-notifications'
+import SettingsContext from '../../../contexts/settings-context'
+import GetApiHandler, { DeleteApiHandler } from '../../../utils/ApiHandler'
+import Button from '../../Common/Button'
+import CreateNotificationModal, {
+  AgentConfiguration,
+} from './CreateNotificationModal'
+import CachedImage from '../../Common/CachedImage'
+
+const NotificationSettings = () => {
+  const settingsCtx = useContext(SettingsContext)
+  const [addModalActive, setAddModalActive] = useState(false)
+  const [configurations, setConfigurations] = useState<AgentConfiguration[]>()
+  const [editConfig, setEditConfig] = useState<AgentConfiguration>()
+  const { addToast } = useToasts()
+
+  const basePath = process.env.NEXT_PUBLIC_BASE_PATH ?? ''
+
+  useEffect(() => {
+    document.title = 'Maintainerr - Settings - Notifications'
+    GetApiHandler('/notifications/configurations').then((configs) =>
+      setConfigurations(configs),
+    )
+  }, [])
+
+  useEffect(() => {
+    GetApiHandler('/notifications/configurations').then((configs) =>
+      setConfigurations(configs),
+    )
+  }, [addModalActive])
+
+  const doEdit = (id: number) => {
+    const config = configurations?.find((c) => c.id === id)
+
+    setEditConfig(config)
+    setAddModalActive(!addModalActive)
+  }
```

Comment:
```suggestion
  const doEdit = (id: number) => {
    const config = configurations?.find((c) => c.id === id)

    setEditConfig(config)
    setAddModalActive(!addModalActive)
    GetApiHandler('/notifications/configurations').then((configs) =>
      setConfigurations(configs),
    )
  }
```

Ref: https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes